packages/flutter/lib/src/material/snack_bar.dart

```diff
@@ -245,15 +250,40 @@ class SnackBar extends StatelessWidget {
     final MediaQueryData mediaQueryData = MediaQuery.of(context);
     assert(animation != null);
     final ThemeData theme = Theme.of(context);
+    final ColorScheme colorScheme = theme.colorScheme;
     final SnackBarThemeData snackBarTheme = theme.snackBarTheme;
-    // TODO(rami-a): Use a light theme if the app has a dark theme, https://github.com/flutter/flutter/issues/31418
-    final ThemeData darkTheme = ThemeData(
-      brightness: Brightness.dark,
-      accentColor: theme.accentColor,
-      accentColorBrightness: theme.accentColorBrightness,
+    final bool isThemeDark = theme.brightness == Brightness.dark;
+
+    // SnackBar uses a theme that is the opposite brightness from
+    // the surrounding theme.
+    final Brightness brightness = isThemeDark ? Brightness.light : Brightness.dark;
+    final Color themeBackgroundColor = isThemeDark
+      ? colorScheme.onSurface
+      : Color.alphaBlend(colorScheme.onSurface.withOpacity(0.80), colorScheme.surface);
+    final ThemeData inverseTheme = ThemeData(
+      brightness: brightness,
+      backgroundColor: themeBackgroundColor,
+      colorScheme: ColorScheme(
```

Comment:
I think you could maybe use `copyWith` here to make this a bit more concise since you only want to change the secondary color and the brightness.